src/Roslyn.Suppressions/Roslyn.Suppressions.UnitTests/SuppressionCheckerTests.cs

```diff
@@ -314,12 +320,25 @@ private static Mock<IChecksumCalculator> CreateChecksumCalculator(string hashToR
             // The issue will match because the line number is the same (so the hash is irrelevant)
             var sonarLine = diagnostic.Location.GetLineSpan().EndLinePosition.Line;
 
-            var issuePath = overrideFilePath ?? diagnostic.Location.SourceTree.FilePath;
-            var issueHash = System.Guid.NewGuid().ToString();
+            var serverIssuePath = overrideFilePath ?? GetMatchingRelativePath(diagnostic.Location.SourceTree.FilePath);
+
+            var issueHash = Guid.NewGuid().ToString();
 
-            return CreateIssue(diagnostic.Id, issuePath, sonarLine, issueHash);
+            return CreateIssue(diagnostic.Id, serverIssuePath, sonarLine, issueHash);
         }
 
+        private static string GetMatchingRelativePath(string absolutePath)
+        {
+            // The matching server should not be rooted, so we'll strip off the
+            // root and return the rest of the path. The product code should be
+            // considered that to be a valid match for the original full path.
+            if (!Path.IsPathRooted(absolutePath))
+            {
+                throw new ArgumentException(nameof(absolutePath), $"Test setup error: path should be rooted. Actual: {absolutePath}");
+            }
+            var root = Path.GetPathRoot(absolutePath);
+            return absolutePath.Substring(root.Length);
+        }
```

Comment:
The `CreateIssueFromDiagnostic` method is a test helper method that creates a matching server issue from a test diagnostic. Previously it was ok if the paths were the same; now the server path has to be relative.